python/cudf_polars/cudf_polars/containers/dataframe.py

```diff
@@ -14,34 +14,33 @@
 
 import polars as pl
 
-from cudf_polars.containers.column import NamedColumn
+from cudf_polars.containers import Column
 from cudf_polars.utils import dtypes
 
 if TYPE_CHECKING:
-    from collections.abc import Mapping, Sequence, Set
+    from collections.abc import Iterable, Mapping, Sequence, Set
 
     from typing_extensions import Self
 
-    from cudf_polars.containers import Column
-
 
 __all__: list[str] = ["DataFrame"]
 
 
 class DataFrame:
     """A representation of a dataframe."""
 
-    columns: list[NamedColumn]
+    column_map: dict[str, Column]
     table: plc.Table
 
-    def __init__(self, columns: Sequence[NamedColumn]) -> None:
-        self.columns = list(columns)
-        self._column_map = {c.name: c for c in self.columns}
-        self.table = plc.Table([c.obj for c in columns])
+    def __init__(
+        self, columns: Iterable[tuple[str, Column]] | Mapping[str, Column]
```

Comment:
Curious if there's a reason a mapping or a iterable of tuples can be accepted here instead of one or the other.

Reply:
I could be convinced of only accepting one or the other, but I found it convenient (I think) to be able to pass both. Although mostly I pass iterables I think.

Reply:
Just asking because IMO it would be nice to accept one type of argument (although without manually checking the type, `dict` would still accept either argument).

An alternative would be to make the constructor be `def __init__(self, columns: Iterable[Column], names: Iterable[str]):`, but I understand if that's a larger refactor than you want to do here.

Reply:
I don't like that as much because it's easier to produce a mismatching length of names and columns.

Reply:
Sounds good. I'm OK with this signature then

Reply:
I looked again and it seems like I only use the `Iterable` constructor, so I'll check if I can remove the `Mapping` case.

Reply:
Done.